utils/sl/prereq_tools/base.py

```diff
@@ -810,6 +812,15 @@ def has_source(self, env, *comps, **kw):
         return True
 
 # pylint: enable=too-many-branches
+
+    def _setup_build_type(self):
+        """set build type"""
+        self.add_opts(EnumVariable('BUILD_TYPE', "Set the build type",
+                                   'dev', ['dev', 'debug', 'release'],
+                                   ignorecase=1))
+
```

Comment:
Does this create a command-line option like AddOption() method does ? And if yes, what will be the option name/syntax then ?
Also why here in prereq_tools module ?

Reply:
Yes.  

scons BUILD_TYPE=release COMPILER=clang will build using build/release/clang as the Variant dir

It is better here in sl because this is where the function is implemented where it gets the build directory from so it seems natural to put it there.

These options are sticky and get saved in daos.conf so you only need do scons on subsequent builds.    You can also switch between different build by changing the value.